react/components/Slide.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import EventListener from 'react-event-listener'
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+import debounce from 'debounce'
+import styles from './styles.css'
+import { NoSSR } from 'vtex.render-runtime'
+
+class SlideComponent extends Component {
+  constructor(props) {
+    super(props)
+    this.imgRef = React.createRef()
+    if (typeof window !== 'undefined') {
+      this.handleResize = debounce(() => {
+        setTimeout(() => {
+          this.fit()
+        }, props.resizeDebounce)
+      }, props.resizeDebounce)
+    }
+  }
+
+  componentDidMount() {
+    this.ensureImageCover()
+  }
+
+  componentDidUpdate() {
+    this.ensureImageCover()
+  }
+
+  componentWillUnmount() {
+    this.handleResize.clear()
+  }
+
+  fit = () => {
+    const { current: imgElement } = this.imgRef
+
+    if (!imgElement || !imgElement.complete) {
+      return
+    }
+
+    if (
+      imgElement.width / imgElement.height >
```

Comment:
Put this condition inside a constant with a good naming would improve readability